lib/MySQL_Session.cpp

```diff
@@ -4968,6 +4962,15 @@ int MySQL_Session::handler() {
 				gtid_hid = -1;
 				if (rc==0) {
 
+					if (active_transactions != 0) {  // run this only if currently we think there is a transaction
+						// Redundant?
+						//if ((myconn->mysql->server_status & SERVER_STATUS_IN_TRANS) == 0) { // there is no transaction on the backend connection
```

Comment:
`bool MySQL_Connection::IsActiveTransaction()` does the same and `unsigned int MySQL_Session::NumActiveTransactions(bool check_savepoint)` calls it for all the backends

Reply:
I didn't understand.

Reply:
I mean that we do not need to do `if ((myconn->mysql->server_status & SERVER_STATUS_IN_TRANS) == 0)` because `MySQL_Session::NumActiveTransactions` already does it for the connection of each backend associated to the session.

Reply:
And I mean that I think we don't need to call `NumActiveTransactions()` if `active_transaction != 0` and `(myconn->mysql->server_status & SERVER_STATUS_IN_TRANS) == 0)` .
Thoughts?

Reply:
You are right. So the objective of this code is to save unneeded fetching of `server_status` of the other connections if we know `myconn->mysql->server_status` is already in a transaction?

Reply:
Right